coffea/processor/executor.py

```diff
@@ -553,23 +596,94 @@ def __call__(
         if self.compression is not None:
             function = _compression_wrapper(self.compression, function)
 
+        def merge_tqdm(chunks, accumulator, desc="Adding"):
+            gen = (c for c in chunks)
+            return accumulate(
+                tqdm(
+                    gen if self.compression is None else map(_decompress, gen),
+                    disable=not self.status,
+                    unit=self.unit,
+                    total=len(chunks),
+                    desc=desc,
+                ),
+                accumulator,
+                )
+
         def processwith(pool):
-            gen = _futures_handler(
-                {pool.submit(function, item) for item in items}, self.tailtimeout
+            reducer = _reduce(self.compression)
+
+            FH = FuturesHolder(
+                set(pool.submit(function, item) for item in items), refresh=2
             )
+
+            if isinstance(self.merging, tuple) and len(self.merging) == 3:
+                nparts, minred, maxred = self.merging
+            else:
+                nparts, minred, maxred = 5, 4, 100
+
             try:
-                return accumulate(
-                    tqdm(
-                        gen if self.compression is None else map(_decompress, gen),
-                        disable=not self.status,
-                        unit=self.unit,
-                        total=len(items),
-                        desc=self.desc,
-                    ),
-                    accumulator,
-                )
-            finally:
-                gen.close()
+                pbar = tqdm(disable=not self.status, unit=self.unit, total=len(items),
```

Comment:
Add context managers for the progress bars (I should have done that before)

Reply:
that's there just to make the merging optional, nesting with with the conditional looks weird